specs/typography.md

```diff
@@ -0,0 +1,177 @@
+
+(**Editorial**: can we do this, or do we need to build this support?)
+
+### ~~API~~
+
+*~~The key elements of the component's public API surface~~*
+
+### Adaptive Styling Properties (**Editorial**: Added this section)
+
+- `type-ramp-base-size` = "14px"  
+The base size for primary body text. Other type ramp sizes may eventually be determined from this starting value.
+
+### Anatomy and Appearance
+
+The type ramp is constructed as css variables that go plus and minus from the base. This allows a consumer to extend the ramp if necessary.
+
+This ramp is currently based on fixed values according to the current visual design, but could be updated from an algorithm if desired.
+
+- `fast-type-ramp-base-size` = `type-ramp-base-size`
+- `fast-type-ramp-plus-1-size` = "16px"
+- `fast-type-ramp-plus-2-size` = "20px"
+- `fast-type-ramp-plus-3-size` = "28px"
+- `fast-type-ramp-plus-4-size` = "34px"
+- `fast-type-ramp-plus-5-size` = "46px"
+- `fast-type-ramp-plus-6-size` = "60px"
+- `fast-type-ramp-minus-1-size` = "12px"
+- `fast-type-ramp-minus-2-size` = "10px"
+
+The semantically named styles that are intended to be consumed at the component level:
+
+- `fast-display-type-size` = `fast-type-ramp-plus-6-size`
+- `fast-header-type-size` = `fast-type-ramp-plus-5-size`
+- `fast-subheader-type-size` = `fast-type-ramp-plus-4-size`
+- `fast-page-title-type-size` = `fast-type-ramp-plus-3-size`
+- `fast-section-title-type-size` = `fast-type-ramp-plus-2-size`
+- `fast-paragraph-larger-type-size` = `fast-type-ramp-plus-1-size`
+- `fast-paragraph-type-size` = `fast-type-ramp-base-size`
+- `fast-paragraph-smaller-type-size` = `fast-type-ramp-minus-1-size`
+- `fast-label-type-size` = `fast-type-ramp-base-size`
+- `fast-button-type-size` = `fast-type-ramp-base-size`
+- `fast-caption-type-size` = `fast-type-ramp-minus-1-size`
+- `fast-caption-smaller-type-size` = `fast-type-ramp-minus-2-size`
+
+The default mappings for standard elements to styles:
+
+- `body` = `fast-paragraph`
+- `h1` = `fast-page-title`
+- `h2` = `fast-section-title`
+- `h3` = `fast-type-ramp-plus-1`
+- `h4` = `fast-type-ramp-base`
+- `h5` = `fast-type-ramp-minus-1`
+- `h6` = `fast-type-ramp-minus-2`
+- `label` = `fast-label`
+
+---
+
+## Implementation
+
+*~~Important aspects of the planned implementation with careful consideration of web standards and integration.~~*
+
+### ~~States~~
+
+*~~Key component states, valid state transitions, and how interactions trigger a state transition.~~*
+
+### Accessibility
+
+The benefit of enabling standard elements to be styled is that if used properly they will already have the appropriate default aria roles and attributes. The downside is potential misuse and missing attributes. For instance ```<p class="fast-heading">My Heading</p>``` won't be properly identified as a heading by assistive technology.
```

Comment:
I'm not sure this assumption is accurate. Just because I want the visual treatment of a fast-heading doesn't mean I want a semantic treatment of that in a given scenario. The same could be said with subheading, paragraph, etc.

Reply:
This *is* accurate if my intent here is that this would be identified as a heading, but I kind of feel like this is a non-goal here. This reads to me as though we're trying to enforce a specific standard rather than support and empower.

Reply:
I agree on empowering developers to easily create accessible markup. My experience has been that separating the semantic markup from the visual treatment of these better empowers developers to do the right thing and create accessible experiences. At the core of this, if the intent is to support and empower accessibility easily I would suggest that our best method of doing that is to keep it simple and simply provide classes as a way to style. This would allow us to point to the standards and recommendations while staying completely out of the way and keeping a singular way of applying the visual system. Doing that does make preventing abuse a non-goal, but I do think it creates clarity.

1. Markup should be semantic and accessible per the existing standards, you can use any elements you need to accomplish that in your app/page/scenario.

2. Classes are the methodology of applying visual styles to your semantic markup.

Again, this does leave room for what we might think is abusing the system (`<p class="my-heading">Foo</p>`), but I don't think we can assume that others would share that exact guidance. If the intended approach is to add more guardrails to prevent misuse I think we are working against our principles.

Reply:
I still believe we can support both, and will include this with examples in the headings spec.